libraries/WebServer/src/WebServer.cpp

```diff
@@ -128,91 +129,185 @@ static String md5str(String &in){
-      if(encoded == NULL){
-        authReq = "";
-        free(toencode);
-        return false;
-      }
-      sprintf(toencode, "%s:%s", username, password);
-      if(base64_encode_chars(toencode, toencodeLen, encoded) > 0 && authReq.equalsConstantTime(encoded)) {
-        authReq = "";
-        free(toencode);
-        free(encoded);
-        return true;
-      }
-      free(toencode);
-      free(encoded);;
-    } else if(authReq.startsWith(F("Digest"))) {
-      authReq = authReq.substring(7);
-      log_v("%s", authReq.c_str());
-      String _username = _extractParam(authReq,F("username=\""),'\"');
-      if(!_username.length() || _username != String(username)) {
-        authReq = "";
-        return false;
-      }
-      // extracting required parameters for RFC 2069 simpler Digest
-      String _realm    = _extractParam(authReq, F("realm=\""),'\"');
-      String _nonce    = _extractParam(authReq, F("nonce=\""),'\"');
-      String _uri      = _extractParam(authReq, F("uri=\""),'\"');
-      String _response = _extractParam(authReq, F("response=\""),'\"');
-      String _opaque   = _extractParam(authReq, F("opaque=\""),'\"');
-
-      if((!_realm.length()) || (!_nonce.length()) || (!_uri.length()) || (!_response.length()) || (!_opaque.length())) {
-        authReq = "";
-        return false;
-      }
-      if((_opaque != _sopaque) || (_nonce != _snonce) || (_realm != _srealm)) {
-        authReq = "";
-        return false;
-      }
-      // parameters for the RFC 2617 newer Digest
-      String _nc,_cnonce;
-      if(authReq.indexOf(FPSTR(qop_auth)) != -1 || authReq.indexOf(FPSTR(qop_auth_quoted)) != -1) {
-        _nc = _extractParam(authReq, F("nc="), ',');
-        _cnonce = _extractParam(authReq, F("cnonce=\""),'\"');
-      }
-      String _H1 = md5str(String(username) + ':' + _realm + ':' + String(password));
-      log_v("Hash of user:realm:pass=%s", _H1.c_str());
-      String _H2 = "";
-      if(_currentMethod == HTTP_GET){
-          _H2 = md5str(String(F("GET:")) + _uri);
-      }else if(_currentMethod == HTTP_POST){
-          _H2 = md5str(String(F("POST:")) + _uri);
-      }else if(_currentMethod == HTTP_PUT){
-          _H2 = md5str(String(F("PUT:")) + _uri);
-      }else if(_currentMethod == HTTP_DELETE){
-          _H2 = md5str(String(F("DELETE:")) + _uri);
-      }else{
-          _H2 = md5str(String(F("GET:")) + _uri);
-      }
-      log_v("Hash of GET:uri=%s", _H2.c_str());
-      String _responsecheck = "";
-      if(authReq.indexOf(FPSTR(qop_auth)) != -1 || authReq.indexOf(FPSTR(qop_auth_quoted)) != -1) {
-          _responsecheck = md5str(_H1 + ':' + _nonce + ':' + _nc + ':' + _cnonce + F(":auth:") + _H2);
-      } else {
-          _responsecheck = md5str(_H1 + ':' + _nonce + ':' + _H2);
-      }
-      log_v("The Proper response=%s", _responsecheck.c_str());
-      if(_response == _responsecheck){
+bool WebServer::authenticateBasicSHA1(const char * _username, const char * _sha1Base64orHex) {
```

Comment:
this method seems to have indentation issues. There are some more throughout the file as well, but fix only what has been changed